subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -51,62 +43,133 @@ enum {
 	ADV_FLAGS_NUM
 };
 
-static struct {
+typedef struct net_buf* (*adv_buf_handler_t)(k_timeout_t timeout);
+
+struct ext_adv {
 	ATOMIC_DEFINE(flags, ADV_FLAGS_NUM);
 	struct bt_le_ext_adv *instance;
 	struct net_buf *buf;
 	uint64_t timestamp;
 	struct k_work_delayable work;
```

Comment:
I feel that only one `k_work_delayable` would be enough for all advertiser. Each advertiser could only keep the next scheduling time and the work could be scheduled at the earliest possible scheduling time. Although the code complexity may increase a bit.

Reply:
It will be complicated, but I think it can be put in the future PR.